versioned/storage/common/src/main/java/org/projectnessie/versioned/storage/common/persist/ObjType.java

```diff
@@ -25,4 +27,21 @@ public interface ObjType {
 
   /** The target class that objects of this type should be serialized from and deserialized to. */
   Class<? extends Obj> targetClass();
+
+  /**
+   * Allows an object type to define how long a particular object instance can be cached.
+   *
+   * <p>{@value #CACHE_UNLIMITED}, which is the default implementation, defines that an object
+   * instance can be cached forever.
+   *
+   * <p>{@value #NOT_CACHED} defines that an object instance must never be cached.
+   *
+   * <p>A positive value defines the maximum age of an object
+   */
+  default long cachedObjectExpiresAtMicros(Obj obj, LongSupplier clock) {
```

Comment:
Would it make sense to declare a duration here, instead of the wall time? This would leave the actual time computation to the caching backend and wouldn't require a clock at all.

Reply:
Wouldn't work for all cases, especially when the object's been read from the database.

Reply:
>  when the object's been read from the database.

Sorry, I'm confused, can you elaborate? On a normal cache setup, the object would be read from the DB, put in the cache, and its expiry time would be = load time + object's cache duration.

Are you implying that objects would have "hard" expiration times, past which they should be evicted _from both the cache and the database_? But if that's the case, shouldn't all `Persist` implementations be modified to check if the fetched object is expired?